### PR TITLE
Add network connected constraint to DataDonationAnalyticsWorker (EXPOSUREAPP-5286)

### PR DESCRIPTION
**Pull request by kolyaopahle**
### Description
This PR adds a network connected constraint to the work builder of the DataDonationAnalyticsWorker as we cannot proceed with analytics donation without a network connection there is no need to enqueue work in that case

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/analytics/worker/DataDonationAnalyticsWorkBuilder.kt [hunk 1]
```diff
@@ -25,5 +27,11 @@ class DataDonationAnalyticsWorkBuilder @Inject constructor() {
                 BackgroundConstants.BACKOFF_INITIAL_DELAY,
                 TimeUnit.MINUTES
             )
+            .setConstraints(buildConstraints())
```
Comment (harambasicluka) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/analytics/worker/DataDonationAnalyticsWorkBuilder.kt:30]: Would it make sense to add a log if the worker wasn't started due to the network constraint?

**Review by chiljamgossow — APPROVED** [review]
LGTM

**Review by mtwalli — APPROVED** [review]
I confirm it is added :)
Screenshot 2021-02-22 at 15 34 49

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/analytics/worker/DataDonationAnalyticsWorkBuilder.kt (hunk 1 above)
Comment (kolyaopahle) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/analytics/worker/DataDonationAnalyticsWorkBuilder.kt:30]: I'm not sure if we can, the work is scheduled by the WorkManager api and if there is no network connection the worker is simply not spawned, i don't think we get any kind of callback

Comment (kolyaopahle) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/analytics/worker/DataDonationAnalyticsWorkBuilder.kt:30]: we could maybe remove the constraint and do the check inside of the worker ourselves but i think this kind of defeats the purpose of the work manager in the first place

**Pull request merged**